src/main/java/at/hannibal2/skyhanni/data/PurseAPI.kt

```diff
@@ -5,16 +5,16 @@ import at.hannibal2.skyhanni.events.LorenzTickEvent
 import at.hannibal2.skyhanni.events.PurseChangeCause
 import at.hannibal2.skyhanni.events.PurseChangeEvent
 import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
-import at.hannibal2.skyhanni.utils.NumberUtil.milion
+import at.hannibal2.skyhanni.utils.NumberUtil.million
 import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
 import net.minecraft.client.Minecraft
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
 
-class PurseAPI {
-    // TODO USE SH-REPO
-    private val pattern = "(Piggy|Purse): §6(?<coins>[\\d,]*).*".toPattern()
-    private var currentPurse = 0.0
+object PurseAPI {
+    val pursePattern by RepoPattern.pattern("data.purse.coins", "(§.)*(Piggy|Purse): §6(?<coins>[\\d,]*(\\.\\d)?)( ?(§.)*\\([+-](?<earned>[\\w,.]+)\\)?|.*)?$")
     private var inventoryCloseTime = 0L
+    var currentPurse = 0.0
```

Comment:
The setter should be private, just to be safe
```suggestion
    var currentPurse = 0.0
      private set;
```